File: BackEnd/Application/UseCases/Projects/SetUsersInProject/SetUsersInProjectCommandHandler.cs
using Domain.Aggregates;
using Infrastructure.Repository;
using Infrastructure.UnitOfWork;
using MediatR;
using Microsoft.EntityFrameworkCore;

namespace Application.UseCases.Projects.SetUsersInProject;

public class SetUsersInProjectCommandHandler(IRepository<User, Guid> userRepository, IRepository<Project, Guid> projectRepository, IUnitOfWork unitOfWork) : IRequestHandler<SetUsersInProjectCommand, bool>
{
    public async Task<bool> Handle(SetUsersInProjectCommand request, CancellationToken cancellationToken)
    {
        var users = await userRepository.Where(u => request.UserIds.Contains(u.Id)).ToListAsync(cancellationToken);
        var project = await projectRepository.Where(p => request.Id.Equals(p.Id)).Include(u => u.Users)
            .FirstOrDefaultAsync(cancellationToken);
        if (project == null) return false;
        project.SetUsers(users);
        await projectRepository.UpdateAsync(project);
        await unitOfWork.SaveChangesAsync(cancellationToken);
        return true;
    }
}

File: flutter/lib/application/use_cases/projects/set_users_in_project/set_users_in_project_command_handler.dart
import 'package:enterprise_management/application/use_cases/projects/set_users_in_project/set_users_in_project_command.dart';
import 'package:enterprise_management/infrastructure/repositories/project_repository.dart';
import 'package:enterprise_management/shared_kernel/cqrs/cqrs.dart';
import 'package:enterprise_management/shared_kernel/result/result.dart';

class SetUsersInProjectCommandHandler extends ICommandHandler<SetUsersInProjectCommand, bool> {
  const SetUsersInProjectCommandHandler({required this._projectRepository});
  final IProjectRepository _projectRepository;

  @override
  Future<Result<bool>> handle(SetUsersInProjectCommand command) async {
    final res = await _projectRepository.setUsersInProject(id: command.id, userIds: command.userIds);
    return Result.success(res);
  }
}